core/src/test/java/org/elasticsearch/search/SearchHitTests.java

```diff
@@ -152,6 +152,33 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * This test adds randomized fields on all json objects and checks that we can parse it to
+     * ensure the parsing is lenient for forward compatibility.
+     * We need to exclude json objects with the "highlight" and "fields" field name since these
+     * objects allow arbitrary keys (the field names that are queries). Also we want to exclude
+     * to add anything under "_source" since it is not parsed, and avoid complexity by excluding
+     * everything under "inner_hits". They are also keyed by arbitrary names and contain SearchHits,
+     * which are already tested elsewhere.
+     */
+    public void testFromXContentLenientParsing() throws IOException {
```

Comment:
I added those as extra test methods so when they fail and the "simple" fromXContent test passes, we know that the problem lies in adding those additional fields. We could randomize it in the existing method, but I like to have one "basic" test without all those extra cases to run with each build as a baseline.  and one for the additional more complicated checks.